KinSDK/KinSDK/Core/KinAccount.swift

```diff
@@ -276,21 +167,42 @@ final class KinStellarAccount: KinAccount {
         }
     }
 
-    func generateTransaction(to recipient: String, kin: Kin, memo: String? = nil, fee: Stroop) -> Promise<TransactionEnvelope> {
+    /**
+     Build a Kin transaction for a specific address.
+
+     - Parameter recipient: The recipient's public address.
+     - Parameter kin: The amount of Kin to be sent.
+     - Parameter memo: An optional string, up-to 28 bytes in length, included on the transaction record.
+     - Parameter fee: The fee in `Quark`s used if the transaction is not whitelisted.
+
+     - Returns: A promise which is signalled with the `Transaction.Envelope` or an `Error`.
+     */
+    public func buildPaymentTransaction(to recipient: String, kin: Kin, memo: String? = nil, fee: Quark) -> Promise<Transaction.Envelope> {
         let txClosure = { (txComp: @escaping GenerateTransactionCompletion) in
-            self.generateTransaction(to: recipient, kin: kin, memo: memo, fee: fee, completion: txComp)
+            self.buildPaymentTransaction(to: recipient, kin: kin, memo: memo, fee: fee, completion: txComp)
```

Comment:
buildPaymentTransaction should return the `PaymentTransaction` object (which can access all transaction fields like hash()), `Transaction.envelope` as far as I see it is an internal object that exposes as a raw string in `WhitlistePayload` and in `BaseTransaction`.
From API consumer point of view, a transaction is an object that has the data and signature.